config/config.go

```diff
@@ -268,6 +276,25 @@ func mergeConfigWithIncludedConfig(config *TerragruntConfig, includedConfig *Ter
 	return includedConfig, nil
 }
 
+// Merge the extra arguments prioritizing those defined in the childExtraArgs
+func mergeExtraArgs(terragruntOptions *options.TerragruntOptions, childExtraArgs []TerraformExtraArguments, parentExtraArgs *[]TerraformExtraArguments) {
+	result := *parentExtraArgs
+addExtra:
+	for _, child := range childExtraArgs {
+		for i, parent := range result {
+			if parent.Name == child.Name {
+				terragruntOptions.Logger.Printf("extra_arguments '%v' from child overriding parent", child.Name)
+				// For extra args, we want to keep the values specified in the child and put them after
+				// the parent ones, so if we encounter a duplicate, we just overwrite it.
+				result[i] = child
+				continue addExtra
```

Comment:
Ah, you're right about the ordering. In that case, instead of a `map`, load them into a `list` if the list doesn't contain the element already, using a method similar to [ListContainsElement](https://github.com/gruntwork-io/terragrunt/blob/master/util/collections.go#L9) to check each item by `Name`. Yes, this is O(n^2), but given that we're dealing with a tiny number of items, it'll make no difference performance-wise, but will make the code easier to read and maintain.